### DISCUSSION TITLE
How to set type for transofrmValues method

### DISCUSSION THREAD

**Discussion by dmitry-ff** [mantinedev/mantine#5643] [Q&A] [answered]
I have a problem with `transformValues` return type. Form type infers from initial form values. And this type is used as return type for `transformValues`  but this method should return another object which is not setisfice with infered form type
How can i resolve this problem?

**Comment by rtivital** [comment, marked as answer]
You can specify form trasformed values type by providing a function type as a second type argument to the useForm hook:

```tsx
intreface FormValues {
  age: string | number;
}

interface TransformedValues {
  age: number;
}

type TransformValues = (values: FormValues) => TransformedValues;

const form  = useForm<FormValues, TransformValues>({
  initialValues: {
    age: '',
  },

  transformValues: values => ({ age: age === '' ? 0 : Number(age) })
})
```
